### PR TITLE
feat: make git requests configurable

### PR DESCRIPTION
**Pull request by alexmt**
PR introduces an `ARGOCD_GIT_REQUEST_TIMEOUT` env variable that allows configuring git request timeouts (currently 15s by default)

### CODE REVIEW

File: util/git/client.go [hunk 1]
```diff
@@ -174,6 +174,10 @@ func NewClientExt(rawRepoURL string, root string, creds Creds, insecure bool, en
 	return client, nil
 }
 
+var (
+	gitClientTimeout = env.ParseDurationFromEnv("ARGOCD_GIT_REQUEST_TIMEOUT", 15, 0, math.MaxInt64)
```
Comment (ishitasequeira) [inline util/git/client.go:178]: Can we add some documentation around the variable? Like add it to the environment variables page?

**Review by ishitasequeira — CHANGES_REQUESTED** [review]
Looks good!! Just one request to add some documentation.

**Comment by crenshaw-dev** [comment]
Should this be configured via a CLI arg/env var default on the repo-server command? Would improve discoverability.

**Review by gdsoumya — APPROVED** [review]

**Comment by alexmt** [comment]
Thank you for review @crenshaw-dev and @ishitasequeira !

I've kept working on debugging the issue caused by the slow git provider and realized another setting is required: `ARGOCD_GIT_LS_REMOTE_PARALLELISM_LIMIT` . Together with `ARGOCD_GIT_REQUEST_TIMEOUT`, these settings helps to address issues caused by overloaded git provider. I've documented both in `argocd-cmd-params-cm` configmap.

PTAL

File: util/git/client.go (hunk 1 above)
Comment (alexmt) [reply util/git/client.go]: done

File: reposerver/metrics/githandlers.go [hunk 2]
```diff
@@ -19,7 +35,13 @@ func NewGitClientEventHandlers(metricsServer *MetricsServer) git.EventHandlers {
 		OnLsRemote: func(repo string) func() {
 			startTime := time.Now()
 			metricsServer.IncGitRequest(repo, GitRequestTypeLsRemote)
+			if lsRemoteParallelismLimitSemaphore != nil {
+				_ = lsRemoteParallelismLimitSemaphore.Acquire(context.Background(), 1)
```
Comment (ishitasequeira) [inline reposerver/metrics/githandlers.go:39]: I think we are ignoring a possible error catch here. Should we be ignoring this?

Comment (alexmt) [reply reposerver/metrics/githandlers.go:41]: The `Acquire` method returns either `nil` or `ctx.Error()`. We are providing `context.Background()` which never get's canceled, so it is safe to ignore error here. Added comment that clarifies it.

**Review by ishitasequeira — APPROVED** [review]
LGTM!!

**Pull request merged**